libraries/mbed/targets/hal/TARGET_Maxim/TARGET_MAX32600/pwmout_api.c

```diff
@@ -77,9 +77,9 @@ void pwmout_init(pwmout_t* obj, PinName pin)
                     } 
 
                     // If all instances are in use, overwrite the last 
-                    pwm = PinMap_PWM[++i];
+                    pwm = PinMap_PWM[i++];
```

Comment:
Well the hardware itself is kind of complicated, so I'm not sure if I can simply the code very much. There are 7 different PWM peripherals that can be mapped to the GPIO pins. Each pin has three that it can use, so there is a chance for a collision between the pins. I will add more comments to the code and try to make it easier to understand.